src/App.jsx

```diff
@@ -13,15 +13,115 @@ function App() {
     const [todoList, setTodoList] = useState([]);
     const [isLoading, setIsLoading] = useState(true);
 
-    useEffect(() => {
-        new Promise((resolve, reject) => {
-            setTimeout(() => {
-                resolve({ data: { todoList: [] } });
-            }, 2000);
-        }).then((result) => {
-            setTodoList(result.data.todoList);
+    // Get tasks from AirTable
+    const fetchData = async () => {
+        const apiURL = `https://api.airtable.com/v0/${
+            import.meta.env.VITE_AIRTABLE_BASE_ID
+        }/${import.meta.env.VITE_TABLE_NAME}`;
+        const options = {
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${
+                    import.meta.env.VITE_AIRTABLE_API_TOKEN
+                }`,
+            },
+        };
+        try {
+            const response = await fetch(apiURL, options);
+
+            if (!response.ok) {
+                const message = `Error has ocurred: ${response.status}`;
+                throw new Error(message);
+            }
+            const data = await response.json();
+            const todos = data.records.map((todo) => {
+                const newTodo = {
+                    id: todo.id,
+                    title: todo.fields.title,
+                };
+                return newTodo;
+            });
+            setTodoList(todos);
             setIsLoading(false);
-        });
+        } catch (error) {
+            console.log(error.message);
+            return null;
+        }
+    };
+
+    // Post a task to Airtable
+    const postTodo = async (todo) => {
+        try {
+            const airtableData = {
+                fields: {
+                    title: todo.title,
+                },
+            };
+
+            const response = await fetch(
+                `https://api.airtable.com/v0/${
+                    import.meta.env.VITE_AIRTABLE_BASE_ID
+                }/Default`,
+                {
+                    method: "POST",
+                    headers: {
+                        "Content-Type": "application/json",
+                        Authorization: `Bearer ${
+                            import.meta.env.VITE_AIRTABLE_API_TOKEN
+                        }`,
+                    },
+                    body: JSON.stringify(airtableData),
+                }
+            );
+
+            if (!response.ok) {
+                const message = `Error has ocurred: ${response.status}`;
+                throw new Error(message);
+            }
+
+            const dataResponse = await response.json();
+            console.log(dataResponse);
+            // return dataResponse;
+            return { id: dataResponse.id, title: dataResponse.fields.title };
+        } catch (error) {
+            console.log(error.message);
+            return null;
+        }
+    };
+
+    // Delete from Airtable
+    const postDeleteTodo = async (id) => {
```

Comment:
This function's name is misleading, we aren't POSTing anything here. We're not sending a body of data to the server, so I think just `deleteTodo` would be more accurate.